src/ImageSharp/Image/PixelAccessor{TColor}.cs

```diff
@@ -186,8 +186,7 @@ public void CopyFrom(PixelArea<TColor> area, int targetY, int targetX = 0)
             int width = Math.Min(area.Width, this.Width - targetX);
             int height = Math.Min(area.Height, this.Height - targetY);
 
-            this.CheckDimensions(width, height);
-
+            // this.CheckDimensions(width, height); TODO: Why was width == 0 or height == 0 considered a problem? Copy implementations do not fail on this (just do nothing)!
```

Comment:
@JimBobSquarePants @dlemstra 
```
int width = Math.Min(area.Width, this.Width - targetX);
int height = Math.Min(area.Height, this.Height - targetY);
```
This will be calculated 2X then. It seems wasteful to me & also seems to be a code smell. It's not the only thing I don't like about the design of the current `PixelArea` <--> `PixelAccessor` and other copying solutions. The switch statements on `ComponentOrder` are very inefficient perf killers.
It's not the time to change the design however, so now I will re-enable the check and duplicate code on the call site, but we should definitely come back to rethink this.